### DISCUSSION TITLE
Feature: Private Vimeo Thumbnails

### DISCUSSION THREAD

**Discussion by ThatGuySam** [ThatGuySam/vimeo-thumbnail-server#20] [Ideas]
The idea is that you can get thumbnails for password-protected videos. 

Here's a theoretical example usage with HTML
```html
<!-- source video: https://vimeo.com/358629078 -->
<img src="https://vumbnail.com/358629078:1234567890.jpg" />
```

**Comment by LukeMeyer94** [comment]
+1 from me, i've built my own regex testing system for deconstructing vimeo links with access codes and reconstructing them for embed

for a little more context, here's the two formats i've seen private video links
```
`https://vimeo.com/${id}/${code}`,
`https://player.vimeo.com/video/${id}?h=${code}`
```

i'm new to open source projects, can i just try and build the solution and open a PR? or do i need to open an issue first

**Reply by ThatGuySam** [reply]
Actually I think what you’re talking about is already implemented here: 

https://github.com/ThatGuySam/vumbnail/issues/9#issuecomment-1076963097

Here’s an example url: 

![](https://vumbnail.com/692839122:3144c3edfc.jpg)

And the docs page: 

https://vumbnail.com/examples/unlisted-vimeo-thumbnails

Let me know if that’s what you’re talking about

**Reply by LukeMeyer94** [reply]
thanks perfect, works great here, thanks

**Reply by ThatGuySam** [reply]
Excellent! 

Be sure to look around and let me know if you have any other Features you’d like to request or +1

**Reply by LukeMeyer94** [reply]
awesome thanks again, this turned a real headache of async vimeo calls into a very simple task
